packages/react-components/react-charts-preview/library/src/utilities/utilities.ts

```diff
@@ -244,42 +257,58 @@ export function createDateXAxis(
   customDateTimeFormatter?: (dateTime: Date) => string,
   useUTC?: string | boolean,
 ) {
-  const { domainNRangeValues, xAxisElement, tickPadding = 6, xAxistickSize = 6, xAxisCount = 6 } = xAxisParams;
+  const {
+    domainNRangeValues,
+    xAxisElement,
+    tickPadding = 6,
+    xAxistickSize = 6,
+    xAxisCount,
+    hideTickOverlap,
+  } = xAxisParams;
   const xAxisScale = useUTC ? d3ScaleUtc() : d3ScaleTime();
   xAxisScale
     .domain([domainNRangeValues.dStartValue, domainNRangeValues.dEndValue])
     .range([domainNRangeValues.rStartValue, domainNRangeValues.rEndValue]);
-  const xAxis = d3AxisBottom(xAxisScale).tickSize(xAxistickSize).tickPadding(tickPadding).ticks(xAxisCount);
-  if (customDateTimeFormatter) {
-    xAxis.tickFormat((domainValue: Date, _index: number) => {
+
+  let tickCount = xAxisCount ?? 6;
+  const tickFormat = (domainValue: Date, _index: number) => {
+    if (customDateTimeFormatter) {
       return customDateTimeFormatter(domainValue);
```

Comment:
we need to ensure user provided customDateTimeFormatter takes priority over other ways to format the ticks